default/python/AI/AIstate.py

```diff
@@ -74,7 +74,12 @@ def convert_to_version(state, version):
 
     if version == 0:
```

Comment:
> Any particular advantage you see in that approach?

It is common practice.   You write updates as separate parts: `a->b`, `b->c`, `c->d`

if you have version `a` you need apply `a->b`, `b->c`, `c->d`
If you have version `b` you need to apply `b->c`, `c->d`

As I see you want to write this by hand.
```python
if version == a:
    a_b()
    b_c()
    c_d()
elif version == b:
    b_c()
    c_d()
```